python/ray/air/execution/resources/resource_manager.py

```diff
@@ -0,0 +1,106 @@
+import abc
+
+from typing import List, Optional
+
+import ray
+from ray.air.execution.resources.request import (
+    ResourceRequest,
+    AllocatedResource,
+)
+from ray.util.annotations import DeveloperAPI
+
+
+@DeveloperAPI
+class ResourceManager(abc.ABC):
```

Comment:
Since this is tightly coupled with the `ResourceRequest` should they be defined in one file? (No strong opinions here)

Reply:
I'd like to keep it separate for better readability